### DISCUSSION TITLE
mac os kitty ssh into linux box

### DISCUSSION THREAD

**Discussion by silicali** [kovidgoyal/kitty#4038] [Q&A] [answered]
I use kitty 0.23.1 (see below for config). My regular work is kitty--ssh---->linux build-server---->tmux-with-multple-session
No issue when connection doesn't timeout but when mac(hibernate) or my ssh timeout I see below message on kitty window.

                      "client_loop: send disconnect: Broken pipe"
but whenever I click on the term it keeps printing below code for every click until I type reset terminal.
                         mac-laptop$ 0;164;37M0;164;37m9;5u

Is there any term settings I need to do or typing "reset" on the kitty terminal is the only way (yes I can close the kitty & restart it).
Looks to me some kind of mouse settings it lost.

Thank you in advance

kitty 0.23.1 created by Kovid Goyal
Darwin mac-laptop 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:21 PDT 2021; root:xnu-7195.141.6~3/RELEASE_X86_64 x86_64
ProductName:    macOS ProductVersion:   11.6 BuildVersion:      20G165
Loaded config files:
   /Users/macuser/Desktop/.config/kitty/kitty.conf

Config options different from defaults:
allow_remote_control      y
bell_on_tab               False
bold_font                 JetBrains Mono Bold
bold_italic_font          IBM Plex Mono Bold Italic
click_interval            0.5
clipboard_control         ('write-clipboard', 'write-primary', 'no-append')
clipboard_max_size        10.0
confirm_os_window_close   1
copy_on_select            clipboard
default_pointer_shape     arrow
editor                    /Users/macuser/Desktop/local/bin/nvim
enable_audio_bell         False
focus_follows_mouse       True
font_family               JetBrains Mono Medium
font_size                 15.0
initial_window_height     (860, 'px')
initial_window_width      (1280, 'px')
italic_font               IBM Plex Mono Medium Italic
macos_option_as_alt       3
macos_thicken_font        0.1
select_by_word_characters :@-./_~?&=%+#
tab_bar_style             powerline
tab_powerline_style       round
url_prefixes              ('http', 'https', 'file', 'ftp', 'irc', 'gopher', 'mailto', 'news', 'git')
url_style                 1
window_padding_width      FloatEdges(left=2.0, top=2.0, right=2.0, bottom=2.0)
Added mouse actions:
         alt+b1 press ungrabbed KeyAction('mouse_selection', (2,))
         alt+b1 press grabbed KeyAction('mouse_selection', (2,))
         cmd+b1 repeat ungrabbed KeyAction('paste_from_selection')
         cmd+b2 press ungrabbed KeyAction('mouse_selection', (1,))
Added shortcuts:
        f6 KeyAction('create_marker')
        shift+enter KeyAction('send_text', ('all', b'\x1b[13;2u'))
        shift+left KeyAction('send_text', ('all', b'\x1b[1;2D'))
        shift+right KeyAction('send_text', ('all', b'\x1b[1;2C'))
        shift+up KeyAction('send_text', ('all', b'\x1b[0H'))
        shift+down KeyAction('send_text', ('all', b'\x1b[0F'))
        shift+f6 KeyAction('remove_marker')
        alt+, KeyAction('scroll_to_mark', (False, 0))
        alt+. KeyAction('scroll_to_mark', (True, 0))
        alt+backspace KeyAction('send_text', ('all', b'\x1b\x7f'))
        ctrl+enter KeyAction('send_text', ('all', b'\x1b[13;5u'))
        ctrl+backspace KeyAction('send_text', ('all', b'\x1b\x7f'))
        cmd+. KeyAction('send_text', ('all', b'\x05'))
        cmd+[ KeyAction('previous_tab')
        cmd+] KeyAction('next_tab')
        cmd+d KeyAction('debug_config')
        cmd+o KeyAction('edit_config_file')
        cmd+s KeyAction('paste_from_selection')
        cmd+u KeyAction('input_unicode_character')
        cmd+x KeyAction('close_tab')
        cmd+delete KeyAction('clear_terminal', ('reset', True))
        cmd+left KeyAction('send_text', ('all', b'\x1bb'))
        cmd+right KeyAction('send_text', ('all', b'\x1bf'))
Changed shortcuts:
        ctrl+tab KeyAction('send_text', ('all', b''))
        cmd+, KeyAction('send_text', ('all', b'\x01'))
        cmd+r KeyAction('load_config_file')
        cmd+up KeyAction('send_text', ('all', b'\x1b[H'))
        cmd+down KeyAction('send_text', ('all', b'\x1b[F'))
:

**Comment by kovidgoyal** [comment, marked as answer]
That means the program you were using on the remote side, has enabled
mouse events (probably tmux). If a connection dies it means that tmux
had no chance to turn off the mouse events, so kitty will keep on
sending them. Yes the only way to recover is to do a reset, which
can be easily done by pressing ctrl+shift+delete

**Comment by silicali** [comment]
Thanks  for the quick reply.  Thank you for "the best easy to configure feature rich terminal ever!!"

**Comment by kovidgoyal** [comment]
On Wed, Sep 15, 2021 at 08:51:14PM -0700, slicali wrote:
> Thanks  for the quick reply.  Thank you for "the best easy to configure feature rich terminal ever!!"

You are most welcome :)
